services/command-router/src/main/java/org/eclipse/hono/commandrouter/CommandConsumerFactory.java

```diff
@@ -13,49 +13,20 @@
 
 package org.eclipse.hono.commandrouter;
 
-import java.time.Duration;
-
 import org.eclipse.hono.client.CommandTargetMapper;
-import org.eclipse.hono.client.ConnectionLifecycle;
-import org.eclipse.hono.client.HonoConnection;
-import org.eclipse.hono.client.SendMessageSampler;
 import org.eclipse.hono.client.ServiceInvocationException;
-import org.eclipse.hono.commandrouter.impl.amqp.ProtonBasedCommandConsumerFactoryImpl;
+import org.eclipse.hono.util.Lifecycle;
 
 import io.opentracing.SpanContext;
 import io.vertx.core.Future;
 
 /**
- * A factory for creating clients for the <em>AMQP 1.0 Messaging Network</em> to
- * receive commands and send responses.
+ * A factory for creating consumers of command &amp; control messages.
  */
-public interface CommandConsumerFactory extends ConnectionLifecycle<HonoConnection> {
-
-    /**
-     * Creates a new factory for an existing connection.
-     *
-     * @param connection The connection to the AMQP network.
-     * @return The factory.
-     * @throws NullPointerException if connection or gatewayMapper is {@code null}.
-     */
-    static CommandConsumerFactory create(final HonoConnection connection) {
-        return create(connection, SendMessageSampler.Factory.noop());
-    }
+public interface CommandConsumerFactory extends Lifecycle {
 
     /**
-     * Creates a new factory for an existing connection.
-     *
-     * @param connection The connection to the AMQP network.
-     * @param samplerFactory The sampler factory to use.
-     * @return The factory.
-     * @throws NullPointerException if connection or gatewayMapper is {@code null}.
-     */
-    static CommandConsumerFactory create(final HonoConnection connection, final SendMessageSampler.Factory samplerFactory) {
-        return new ProtonBasedCommandConsumerFactoryImpl(connection, samplerFactory);
-    }
-
-    /**
-     * Initializes the ProtocolAdapterCommandConsumerFactory with the given components.
+     * Initializes the CommandConsumerFactory with the given commandTargetMapper.
```

Comment:
I do not really see the value of the initialize method. It only seems to make implementation harder. Why don't we simply pass in the mapper in the implementation class's constructor? The factory interface itself doesn't require the mapper anyway ...

Reply:
That will be a bigger change. I'd rather do this in a separate commit/PR.

Reply:
ok